### PR TITLE
add MRation script

### PR DESCRIPTION
**Pull request by erezh31**

### CODE REVIEW

File: Scripts/script-IndicatorMaliciousRatioCalculation.yml [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+  description: Max number of incidents to query.
+  defaultValue: "100000"
+- name: maxIndicators
+  description: Max number of indicators to query.
+  defaultValue: "100000"
+- name: appearsInMinNumberOfIncidents
+  description: Indicator appears in minimum number of incidents.
+  defaultValue: "5"
+- name: query
+  description: The query to fetch the indicators
```
Comment (meirwah) [inline Scripts/script-IndicatorMaliciousRatioCalculation.yml]: The query to fetch the indicators from incident which are "bad"

File: Scripts/script-IndicatorMaliciousRatioCalculation.yml [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+args:
+- name: maxIncidents
+  description: Max number of incidents to query.
+  defaultValue: "100000"
+- name: maxIndicators
+  description: Max number of indicators to query.
+  defaultValue: "100000"
+- name: appearsInMinNumberOfIncidents
+  description: Indicator appears in minimum number of incidents.
+  defaultValue: "5"
```
Comment (meirwah) [inline Scripts/script-IndicatorMaliciousRatioCalculation.yml]: change to 3

File: Scripts/script-IndicatorMaliciousRatioCalculation.yml [hunk 3]
```diff
@@ -0,0 +1,110 @@
+commonfields:
```
Comment (meirwah) [inline Scripts/script-IndicatorMaliciousRatioCalculation.yml]: add fromVersion: 3.5.0

File: Scripts/script-IndicatorMaliciousRatioCalculation.yml [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+  def get_incident_data(incident):
+      result = {k: v for k,v in incident.items() if k in ['id', 'type', 'occurred', 'CustomFields', 'labels']}
+      result['labels'] = hash_object(get_incident_labels_map(result.get('labels')))
+      result['CustomFields'] = hash_object(result.get('CustomFields'))
+      return result
+
+
+  res = demisto.executeCommand("findIndicators", {'query': QUERY, 'size': MAX_INDICATORS})
+  indicators = res[0]['Contents']
+  indicators_result = map(get_indiator_data, indicators)
```
Comment (meirwah) [inline Scripts/script-IndicatorMaliciousRatioCalculation.yml]: typo in indiator

File: Scripts/script-IndicatorMaliciousRatioCalculation.yml [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+  res = demisto.executeCommand("getIncidents", {'query': QUERY.replace('incident.', ''), 'size': MAX_INCIDENTS})
```
Comment (meirwah) [inline Scripts/script-IndicatorMaliciousRatioCalculation.yml]: better to query all incidents that have the indicators ids found in indicators_result
More efficient and correct

File: Scripts/script-IndicatorMaliciousRatioCalculation.yml [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+      non_resolved_incidents = map(get_incident_data, non_resolved_incidents)
+
+      indicators_result = [x for x in indicators_result if x['total_incidents_count'] >= APPEARS_IN_MIN_NUMBER_OF_INCIDENTS]
+      indicators_result.sort(key=lambda x: x['malicious_ratio'], reverse=True)
+      demisto.results({
+          'Type': entryTypes['note'],
+          'Contents': indicators_result,
+          'ContentsFormat': formats['json'],
+          'ReadableContentsFormat': formats['markdown'],
+          'HumanReadable': tableToMarkdown('Indicators Malicious Ratio', indicators_result[:50], headers=['value', 'indicator_type', 'malicious_ratio', 'total_incidents_count', 'score', 'lastSeen'])
```
Comment (meirwah) [inline Scripts/script-IndicatorMaliciousRatioCalculation.yml]: make the 50 configurable too as arg

File: Scripts/script-IndicatorMaliciousRatioCalculation.yml [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+      })
+
+      for i in indicators_result:
+          i['value'] = hash_object(i['value'])
+
+      demisto.results(fileResult('MaliciousRatio.json', json.dumps({
```
Comment (meirwah) [inline Scripts/script-IndicatorMaliciousRatioCalculation.yml]: make this configurable by arg.. default should be false (don't generate a file result)

Comment (erezh31) [reply Scripts/script-IndicatorMaliciousRatioCalculation.yml]: I think the default should be yes... with the file result we can calculate more statistics

File: Scripts/script-IndicatorMaliciousRatioCalculation.yml (hunk 1 above)
Comment (erezh31) [reply Scripts/script-IndicatorMaliciousRatioCalculation.yml]: V

File: Scripts/script-IndicatorMaliciousRatioCalculation.yml (hunk 2 above)
Comment (erezh31) [reply Scripts/script-IndicatorMaliciousRatioCalculation.yml]: V

File: Scripts/script-IndicatorMaliciousRatioCalculation.yml (hunk 3 above)
Comment (erezh31) [reply Scripts/script-IndicatorMaliciousRatioCalculation.yml]: V

File: Scripts/script-IndicatorMaliciousRatioCalculation.yml (hunk 4 above)
Comment (erezh31) [reply Scripts/script-IndicatorMaliciousRatioCalculation.yml]: V

File: Scripts/script-IndicatorMaliciousRatioCalculation.yml (hunk 5 above)
Comment (erezh31) [reply Scripts/script-IndicatorMaliciousRatioCalculation.yml]: I have to do it in this way so I can divide the incidents to resovled and non-resolved. otherwise I will get all the incidents as a result and it will hard to parse it (the fields to identify if it's resolved or not is actually in the query).
Plus I check the efficient - it's really really slow to query by indicator.id - I guess it's cuz it's a long "or" query and it's not efficient

File: Scripts/script-IndicatorMaliciousRatioCalculation.yml (hunk 6 above)
Comment (erezh31) [reply Scripts/script-IndicatorMaliciousRatioCalculation.yml]: V

**Comment by erezh31** [comment]
fixes:
https://github.com/demisto/content/pull/1497
